state-chain/pallets/cf-swapping/src/tests.rs

```diff
@@ -1934,3 +1985,61 @@ fn broker_bps_is_limited() {
 		);
 	});
 }
+
+#[test]
+fn swaps_are_executed_according_to_execute_at_field() {
```

Comment:
We should add a test for failure too. 
That a) the state is rolled back and no state changes have occurred
and b) that success on the next block they go through, as well as the subsequent block's swaps being attempted

Reply:
The ability to process swaps for blocks other than the current was already covered by safe mode tests, but I now added `swaps_get_retried_on_next_block_after_failure` to test for swaping errors explicitly.